### PR TITLE
pre-alpha - [S-114659] - Added feature to list the workflow catalog in backstage.

### PR DESCRIPTION
**Pull request by sishwarya**
## Release Checklist

- All steps below must be completed prior to merging.

## Changes Made

1. Listing workflow catalog.
2. Listing categories
3. Permissions feature to view plugin
4. Multiple Instance support
5. Apply filters based on categories, Authored by, and Search input.
6. Free text search on title and description.

### Required

- [x] PR title should follow correct format.
  - MAJOR Release is used when there are breaking changes involved
  - MINOR Release is used when enhancements are introduced
  - PATCH Release is used for defects, security, tech chores, etc.
  - pre-alpha is used for testing releases
  - Note:
    - Based on the commit messages, increment the version from the latest release.
    - Push the bumped npm version in package.json back into the repo.
    - Push a tag for the new version back into the repo.
    - [For more details on versioning](https://github.com/phips28/gh-action-bump-version/blob/master/README.md) 
- [x] Checklist of **changes made** added to PR description
- [x] Related issue(s) linked to PR

 
## Reviewer Checklist

- [ ] Author checklist has been reviewed?
- [ ] All acceptance criteria has been met from linked issue
- [ ] Run code locally and verify all changes

## Related Issue(s)

Resolves #XXXX
Story- [E-17885]: Support workflows in Backstage plugins
1. S-114659 - [Implementation] Create a workflow  Screen with mock data
2. S-114660 - [Implementation] Get and Display  Categories [FE & BE]
3. S-114661 - [Implementation] Get Required Workflow Data from Release. [BE]
4. S-116150 - Backstage Release: Workflows FrontEnd BackEnd Integration
5. S-114995 - [Implementation] Overall Release Workflow Catalog: Filters: SearchBy, AuthoredBy, Categories.

Defect-######

### Features
- Listing workflow catalog.
- Listing categories.
- Permissions feature to view plugin.
- Multiple Instance support.
- Apply filters based on categories, Authored by, and Search input.
- Free text search on title and description.
### Features
- API to get release workflow catalog.
- API to get configured release instances.
- API to get release categories 
- Support for DAI release 23.3,  24.1 and 24.3 versions.
- Permissions feature to access backend API's
- Backend API support to apply filters on categories, authored by, and search input.
### Features
- Added support for release workflow catalog.

### CODE REVIEW

File: plugins/dai-release/src/api/DaiReleaseApiClient.test.ts [hunk 1]
```diff
@@ -204,6 +206,175 @@ describe('ReleaseApiClient', () => {
...
+        0,
+        '',
+        [],
+        '',
+        'default',
+      );
+      expect(response).toEqual(workflowCatalogsList);
+    });
+
+    // these cases will be verified after api implementation
```
Comment (nitheshrayuduv) [inline plugins/dai-release/src/api/DaiReleaseApiClient.test.ts:318]: post api implementation we can add this right ?

**Review by nitheshrayuduv — APPROVED** [review]
Looks good to me , if required, we need to add these data into the common changelog ?
release-common changelog
### Features
- Added support for release Workflows.

File: plugins/dai-release-common/CHANGE_LOG.md [hunk 2]
```diff
@@ -3,13 +3,15 @@
...
+
 - Updated dependencies
   - @backstage/plugin-catalog-common@^1.0.26
   - @backstage/plugin-permission-common@^0.8.1
   - @backstage/plugin-permission-node@^0.8.2
+
 ### Features
-- Added support for release Template.
 
+- Added support for release Template.
```
Comment (nitheshrayuduv) [inline plugins/dai-release-common/CHANGE_LOG.md:14]: Should we add workflows ? or it will be generated automatically ?

Comment (nitheshrayuduv) [reply plugins/dai-release-common/CHANGE_LOG.md:14]: I mean Categories and workflow common consts.

**Review by nitheshrayuduv — APPROVED** [review]
Looks good to me :)

File: .github/workflows/version_changelog.yaml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PACKAGEJSON_DIR: ${{ inputs.plugin-path }}
+        with:
+          commit-message: "CI: bumps version to {{version}} [skip ci]"
+          tag-prefix:  '${{ inputs.plugin }}/v'
+          major-wording: 'MAJOR'
+          minor-wording: 'MINOR'
+          preid: 'alpha'
+          target-branch: 'S-114659-GA' # need to map it to main branch
```
Comment (nitheshrayuduv) [inline .github/workflows/version_changelog.yaml:41]: Need to change the target branch

File: .github/workflows/version_changelog.yaml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      - name: Commit and Push Updated Changelogs
+        id: commit-changelog
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+          git add ${{ inputs.plugin-path }}/CHANGE_LOG.md
+          git commit -m "Update ${{ inputs.plugin }}CHANGE_LOG.md files"
+          git push origin 'S-114659-GA' # need to map it to main branch
```
Comment (nitheshrayuduv) [inline .github/workflows/version_changelog.yaml:69]: here need to change the push to.

Comment (Karthigaiselvan) [inline .github/workflows/version_changelog.yaml:69]: Can you update this with master branch in all places

File: .github/workflows/version_changelog.yaml (hunk 3 above)
Comment (sishwarya) [reply .github/workflows/version_changelog.yaml]: deleted the redunant file.

File: .github/workflows/version_changelog.yaml (hunk 4 above)
Comment (sishwarya) [reply .github/workflows/version_changelog.yaml]: deleted the redunant file.

Comment (sishwarya) [reply .github/workflows/version_changelog.yaml]: deleted the redunant file.

**Pull request merged**